aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -256,5 +401,48 @@ Tensor& bernoulli_scalar_cuda_(Tensor &self, double p, Generator* gen) {
   return self;
 }
 
+static void uniform_kernel_cuda(TensorIterator& iter, double from_, double to_, Generator* gen) {
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "uniform_cuda", [&] {
+    auto from = static_cast<scalar_t>(from_);
+    auto to = static_cast<scalar_t>(to_);
+    AT_CHECK(from <= to,
+      "uniform_ expects to return a [from, to) range, but found from=", from,
+      " > to=", to);
+    AT_CHECK((to - from) <= std::numeric_limits<scalar_t>::max(),
+          "uniform_ expects to-from ≤ std::numeric_limits<double>::max(), but found to=", to,
+          " and from=", from, " which result in to-from to exceed the limit");
+
+    using accscalar_t = at::acc_type<scalar_t, true>;
+    auto range = static_cast<accscalar_t>(to-from);
+    from = static_cast<accscalar_t>(from);
+    // define lambda to reverse bounds, multiply 'range' and add 'from_'
+    auto uniform_func = [range, from] __device__ (accscalar_t rand) {
+      // reverse the bounds of curand4 from (0, 1] to [0, 1)
+      // Note that this method is from legacy THCTensorRandom and is likely to give
```

Comment:
Noob question: I thought you were changing the RNG algorithm in this PR, so what's the business with "legacy THCTensorRandom"?

Reply:
Syed is ooto, so filling in for him. It does not matter on what underlying algorithm is used to produce random bits (we are changing it here, from mtgp to philox), you still need a way to transform random bits into a uniformly distributed float. Part of this transformation is done in curand_uniform, that produces (0,1] float/double. In pytorch, however, we need [0,1), and it is slightly tricky to get. Currently (legacy) THCTensorRandom is using a slightly distorting way of doing so (all 1's are turned into 0's), and, in the interest of minimizing changes, we just follow this behavior here.